### ISSUE TITLE
Add the go vendor experiment support

### ISSUE THREAD

**Issue by solher** [go-swagger/go-swagger#226]
Currently, when I try to `swagger generate spec` with the GO15VENDOREXPERIMENT and all my dependencies installed in the `vendor` directory, go-swagger seems to ignore it and crashes during the generation, saying that dependencies doesn't exists.

Is that a known problem or did I miss something ?

Thanks

**Comment by casualjim** [comment]
your vendoring manager should flatten this out. 
go swagger already uses GO15VENDOREXPERIMENT

am I misreading your question?

**Comment by solher** [comment]
Just to be sure there is no misunderstanding, there is no problem running the `swagger` command itself.
There is also no problem doing a `go build`. Everything works fine.

The problem is that go-swagger doesn't seems to look for dependencies inside the vendor directory when generating the spec. It crashes when a `go build` passes under the same conditions.

**Comment by casualjim** [comment]
if your loader project up to date? Does your project compile?

swagger uses the go build infrastructure, We can check out if there is something that is required to be enable.

**Comment by solher** [comment]
Yes it builds. That's the weird thing.
I'm gonna double check if the env var is overrided somewhere when calling the swagger gen...

**Comment by casualjim** [comment]
I've updated the dependencies, there were changes to the golang tools around loader but i don't think it will fix it yet. 

Do you have a repo i can use to reproduce this issue?

**Comment by serussell** [comment]
I also have this problem:

**_go-swagger version (there being no "version" command in the executable)_**
```
1258)go-swagger/go-swagger % git log -1
commit b4422d1e33b980d9653ee089a12593168efcb7b4
Author: Ivan Porto Carrero <ivan@flanders.co.nz>
Date:   Sat Jan 30 22:49:08 2016 -0800

    update code of conduct to 1.4 version
1259)go-swagger/go-swagger % cd -
~/workspace/src/bitbucket.org/internationalsos/riskratings
```
**_The project builds, so Go vendoring is  working:_**
```
1260)internationalsos/riskratings % GO15VENDOREXPERIMENT=1 go build -o build/web ./cmd/web
1261)internationalsos/riskratings % echo $?
0
```
_**To prove that a package is, indeed, in the vendor directory:**_
```
1262)internationalsos/riskratings % ls vendor/github.com/namsral/flag
examples         export_test.go  flag_test.go  README.md
example_test.go  flag.go         LICENSE       testdata
```
**_and that we're using Go 1.5_**
```
1263)internationalsos/riskratings % go version
go version go1.5.2 linux/amd64
```
_**However, go-swagger does not find the vendored packages, and does not appear
to be looking in the ./vendor directory:**_
```
1264)internationalsos/riskratings % GO15VENDOREXPERIMENT=1 swagger generate spec -b ./cmd/web -o swagger.json 2>&1 | head -n 10
/home/ser/workspace/src/bitbucket.org/internationalsos/riskratings/cmd/web/main.go:30:2: could not import github.com/namsral/flag (cannot find package "github.com/namsral/flag" in any of:
        /home/ser/Software/go/src/github.com/namsral/flag (from $GOROOT)
        /home/ser/workspace/src/github.com/namsral/flag (from $GOPATH))
/home/ser/workspace/src/bitbucket.org/internationalsos/riskratings/cmd/web/main.go:33:2: could not import github.com/fvbock/endless (cannot find package "github.com/fvbock/endless" in any of:
```

... and so on, for all dependencies.  Note, mainly, that `vendor` isn't in the search path.  

I can't make this project public, but I'll try to put together a minimum test case for you.

**Comment by casualjim** [comment]
Thank your for this very complete story.
Can you also try to upgrade to go 1.5.3 as there were fixes in this area IIRC.
How are you adding things to your vendor folder, are you using a particular tool?

I'll figure something out for supplying pre-built binaries on github and/or bintray as well as a docker container.

I actually have a similar problem in go-swagger itself where I need to get 3 different dependencies into the gopath but it will happily find all other things.

https://github.com/go-swagger/go-swagger/issues/246
https://github.com/go-swagger/go-swagger/issues/247

**Comment by serussell** [comment]
Hi,

I cleaned and rebuilt swagger using go1.5.3, and it is working now.  Thanks for the pointer!

**Comment by casualjim** [comment]
ok so the problem goes away after a complete clean?

@solher can you close this issue, if the same fix works for you?

**Comment by serussell** [comment]
@casualjim After I upgraded to 1.5.3 and recompiled go-swagger with that version, it started working.

**Comment by solher** [comment]
Hum... After recompiling and running `go build; swagger generate spec;` in a fresh 1.5.3 golang container, I still get the issue while the build succeeds. I'm using Glide btw.

Nevermind, it must be a problem from somewhere else if it fixes the issue for everybody ;)

**Comment by casualjim** [comment]
I believe glide works with ./vendor/src doesn't it?

When i tried glide I had all kinds of issues, so I started using gvt instead, but ymmv

**Comment by solher** [comment]
It doesn't seems like it. The dependencies are directly in ./vendor (e.g. ./vendor/github.com/go-swagger/go-swagger).
I'll give a try to gvt.
Dependency management is so annoying right now. I hope Go 1.6 will help with that.

**Comment by serussell** [comment]
@solher, it may not be you.  Try to clear out all your `$GO*` environment variables and set them to sane values.  I had to mess around a lot to get go-swagger to do vendoring correctly, and I'm not sure exactly what it was I did to make it work.  I reset a lot of my env vars, I installed Go 1.5.3, and I recompiled both `gvt` and `go-swagger` with the newer Go before `go-swagger` worked.  Try a few more things before you give up.

I've been using `gvt` and have been happy with it, BTW.
